src-tauri/src/plugins/cover.rs

```diff
@@ -0,0 +1,91 @@
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use base64::prelude::*;
+use lofty::{MimeType, TaggedFileExt};
+use tauri::plugin::{Builder, TauriPlugin};
+use tauri::Runtime;
+
+use crate::libs::error::AnyResult;
+use crate::libs::utils::scan_dir;
+
+const SUPPORTED_COVER_EXTENSIONS: [&str; 5] = ["png", "jpg", "jpeg", "bmp", "gif"];
+const SUPPORTED_COVER_NAMES: [&str; 5] = ["album", "albumart", "folder", "cover", "front"];
+
+#[memoize::memoize]
+fn get_cover_from_id3(path: String) -> Option<String> {
+    let tagged_file = match lofty::read_from_path(path) {
+        Ok(tagged_file) => tagged_file,
+        Err(_) => return None,
+    };
+
+    let primary_tag = match tagged_file.primary_tag() {
+        Some(tag) => tag,
+        None => return None,
+    };
+
+    let cover = match primary_tag.pictures().first() {
+        Some(cover) => cover,
+        None => return None,
+    };
```

Comment:
I'd recommend doing `primary_tag.get_picture_type(PictureType::FrontCover)` before falling back to whatever the first picture happens to be.